### PR TITLE
Feature automated tests

### PR DESCRIPTION
**Pull request by vlorinc**
Added automated tests of gRPC client functions.
Fixed nil pointer dereference bugs.

Adjusted pull request code test to include dpservice.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ on:
 jobs:
   checks:
     name: run
-    runs-on: ubuntu-latest
+    runs-on: self-hosted
```
Comment (guvenc) [inline .github/workflows/test.yml:13]: We should actually use 
`runs on [self-hosted, dpdk]`
Please check how this is done in metalnet and please do in exact same way.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -19,4 +19,11 @@ jobs:
         uses: actions/setup-go@v5
         with:
           go-version-file: 'go.mod'
+      - name: Run dpservice
+        run: docker run --rm --entrypoint ./dp_service.py --privileged -p1337:1337 --mount type=bind,source=/dev/hugepages,target=/dev/hugepages ghcr.io/ironcore-dev/dpservice-tester --no-init &
```
Comment (guvenc) [inline .github/workflows/test.yml:23]: It should use tester image with `main` label. 

> ghcr.io/ironcore-dev/dpservice-tester:main

I changed this also in metalnet.

**Review by guvenc — CHANGES_REQUESTED** [review]
Thanks. Looks pretty good. Some comments though.

**Comment by vlorinc** [comment]
I changed the test file as requested and also reordered the steps, because dpservice was not always started before `make test` was run.

File: .github/workflows/test.yml (hunk 2 above)
Comment (vlorinc) [reply .github/workflows/test.yml]: changed as requested

File: client/suite_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2022 The Metal Authors.
```
Comment (guvenc) [inline client/suite_test.go:2]: We dont use this license text anymore. Could you please adjust it similar to the other files ?

File: client/client_test.go [hunk 4]
```diff
@@ -0,0 +1,1220 @@
+/*
+Copyright 2022 The Metal Authors.
```
Comment (guvenc) [inline client/client_test.go:2]: Same here.

**Review by guvenc — CHANGES_REQUESTED** [review]
Your change was good enough regarding the "reordering" of the steps in github action but to be on the safe side, I have put an intermediate step which waits for the tcp port of dpservice with a timeout of 10 seconds. This can be added to metalnet as well.

There are sill some license relevant issues, which I didnt notice during my last review.

File: client/suite_test.go (hunk 3 above)
Comment (vlorinc) [reply client/suite_test.go]: License text updated

File: client/client_test.go (hunk 4 above)
Comment (vlorinc) [reply client/client_test.go]: License text updated

**Review by guvenc — APPROVED** [review]
Thanks.

**Pull request merged**